Web3Box

web3andrew opened the pull request:
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/w3f/Grants-Program?pullRequest=1256) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/w3f/Grants-Program?pullRequest=1256) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/w3f/Grants-Program?pullRequest=1256) it.</sub>

semuelle commented on applications/Web3Box.md:
```diff
@@ -0,0 +1,142 @@
+ - Web3 information source customization
+
+#### Web3Box VS Browser ####
+![](https://lh3.googleusercontent.com/x5iDstqgO_-ljPxqvNfHR07JF0Bo8xl1aiBEVKxOfxNqXrydZW2drKlkQQU5oiviknzsgqvGGD90k-UlT4Y5TbZn9c4f7-pOfIanXLSuXeb_svjQOMrMcIGddV2VY5m6MohVoLn2rwBNW85QcHZyYy7GMqWVu-xKt_QbRJ6gQCj4Rg1mlISImyXhd3fsaw)
```
What does it mean that "Rights" are "de-centralised"?

web3andrew replied:
Web3Box is open source and will provide user-friendly customized plug-in features in the future, which are related to the Web3 Tool/Feed aggregation module.

semuelle commented on applications/Web3Box.md:
```diff
@@ -0,0 +1,142 @@
+### Project Details
+
+#### Web3Box Architecture ####
+**![](https://lh3.googleusercontent.com/tlealryMkOZD0i7dK9LcU_TTsbO5J_13YcgeOl8hWI56sLVtnyYEDDY-dCeYLVa7_i4k4PIdHCkNqTjA5BFtqYKBJJMUh5jkklAxK9-9PKljwMfoq2a0l-WoMXgTRToE5Pih0m2qGXUoi_Yjl7dSNIo3J_3uAGBW4qaVxN3TUGKjQ88VxuhntBRcddz6JA)**
```
1. What are the white and green lines? 
2. What feed? What tools?

web3andrew replied:
Q1:
White lines mean three key modules, Privacy-safe Box, Risk Assessment and Web3 Tool/Content Aggregation.
Green lines mean that,  In addition to the dashboard & wallet, the derivative functions of Privacy-safe box will be developed by the follow-up milestone.

Q2:
Tools mean web3 tools navigation, including mirror, medium, mask, etc.
<img width="605" alt="image" src="https://user-images.githubusercontent.com/117150831/200759211-e5dcaec2-757a-4f69-b4e7-7a6cec4b6e87.png">

Feeds mean web3 content aggregation, like RSS 3, Coindesk, DeFiLlama, Messari, etc.
<img width="599" alt="image" src="https://user-images.githubusercontent.com/117150831/200759293-603512b0-7416-4300-98ae-8a101c5b10e4.png">

semuelle commented on applications/Web3Box.md:
```diff
@@ -0,0 +1,142 @@
+
+### Overview
+
+**Web3Box is committed to creating a Web3 UGA platform with a secure environment worldwide**
```
What does UGA stand for? How are you planning to implement this `secure environment`?

web3andrew replied:
UGA stands for Users Generated  Assets, and the standalone local running environment - Win & Mac Web3Box desktop clients are for secure environment.

semuelle reviewed (changes requested):
Hi @web3andrew. Thank you for the application. I left a few questions below. Broadly speaking, there are several big issues with this:

1. There is a mismatch between the architecture diagrams and the deliveries. You are suggesting to include a lot of things, but there are no matching deliverables. If you want to fund parts of your project with a grant, please keep the application's focus on those parts. Also, please keep in mind that the deliverables still need to be useful on their own.
2. There is a lack of technical details. What tech stack are you planning to use? What makes your sandbox secure? Where do the asset and transaction data come from?
3. Since we cannot sign off on a milestone with cost "TBD", it's probably best to remove it. You could apply for a follow-up grant if and when the first grant is delivered and accepted.
4. There is no activity on your or your organisation's Github accounts. It would really help if you could show some experience with building and maintaining open source software and Substrate- or blockchain-related projects.

web3andrew commented:
> Hi @web3andrew. Thank you for the application. I left a few questions below. Broadly speaking, there are several big issues with this:
> 
>     1. There is a mismatch between the architecture diagrams and the deliveries. You are suggesting to include a lot of things, but there are no matching deliverables. If you want to fund parts of your project with a grant, please keep the application's focus on those parts. Also, please keep in mind that the deliverables still need to be useful on their own.
> 
>     2. There is a lack of technical details. What tech stack are you planning to use? What makes your sandbox secure? Where do the asset and transaction data come from?
> 
>     3. Since we cannot sign off on a milestone with cost "TBD", it's probably best to remove it. You could apply for a follow-up grant if and when the first grant is delivered and accepted.
> 
>     4. There is no activity on your or your organisation's Github accounts. It would really help if you could show some experience with building and maintaining open source software and Substrate- or blockchain-related projects.

@semuelle  Thanks 

> Hi @web3andrew. Thank you for the application. I left a few questions below. Broadly speaking, there are several big issues with this:
> 
>     1. There is a mismatch between the architecture diagrams and the deliveries. You are suggesting to include a lot of things, but there are no matching deliverables. If you want to fund parts of your project with a grant, please keep the application's focus on those parts. Also, please keep in mind that the deliverables still need to be useful on their own.
> 
>     2. There is a lack of technical details. What tech stack are you planning to use? What makes your sandbox secure? Where do the asset and transaction data come from?
> 
>     3. Since we cannot sign off on a milestone with cost "TBD", it's probably best to remove it. You could apply for a follow-up grant if and when the first grant is delivered and accepted.
> 
>     4. There is no activity on your or your organisation's Github accounts. It would really help if you could show some experience with building and maintaining open source software and Substrate- or blockchain-related projects.

@semuelle Thanks, sir, we will update what you metioned asap.

web3andrew commented:
> Hi @web3andrew. Thank you for the application. I left a few questions below. Broadly speaking, there are several big issues with this:
> 
>     1. There is a mismatch between the architecture diagrams and the deliveries. You are suggesting to include a lot of things, but there are no matching deliverables. If you want to fund parts of your project with a grant, please keep the application's focus on those parts. Also, please keep in mind that the deliverables still need to be useful on their own.
> 
>     2. There is a lack of technical details. What tech stack are you planning to use? What makes your sandbox secure? Where do the asset and transaction data come from?
> 
>     3. Since we cannot sign off on a milestone with cost "TBD", it's probably best to remove it. You could apply for a follow-up grant if and when the first grant is delivered and accepted.
> 
>     4. There is no activity on your or your organisation's Github accounts. It would really help if you could show some experience with building and maintaining open source software and Substrate- or blockchain-related projects.

@semuelle  Thanks, sir, we will update what you metioned asap.

web3andrew commented:
> Hi @web3andrew. Thank you for the application. I left a few questions below. Broadly speaking, there are several big issues with this:
> 
>     1. There is a mismatch between the architecture diagrams and the deliveries. You are suggesting to include a lot of things, but there are no matching deliverables. If you want to fund parts of your project with a grant, please keep the application's focus on those parts. Also, please keep in mind that the deliverables still need to be useful on their own.
> 
>     2. There is a lack of technical details. What tech stack are you planning to use? What makes your sandbox secure? Where do the asset and transaction data come from?
> 
>     3. Since we cannot sign off on a milestone with cost "TBD", it's probably best to remove it. You could apply for a follow-up grant if and when the first grant is delivered and accepted.
> 
>     4. There is no activity on your or your organisation's Github accounts. It would really help if you could show some experience with building and maintaining open source software and Substrate- or blockchain-related projects.
@semuelle  Answers are as follows, and updated the application MD file https://github.com/w3f/Grants-Program/pull/1256/commits/e76de15433e3dcd957fc6896179dd813ddc2c1fb
Q1:
We deleted milestone2 and will focus on milestone1 (Dashboard & Multi-chain Wallet). We can make sure that Dashboard & Multi-chain Wallet would be useful after we finish milestone1. and then we will apply for the follow-up Milestone2 Risk Assessment for parachains.
Q2:
Web3Box Wins and Mac Clients will be provided after finishing Milestone1 
Client language: React， Electron
Server language: Nodejs
Q3:
Deleted the Milestone2, and will continue to apply after finishing Milestone1
Q4:
ok, sir, we will update some work that we have done in our official GitHub account these days  https://github.com/web3box-labs/apps

web3andrew commented:
@semuelle   Sir，already updated our [Web3Box Github](https://github.com/web3box-labs/apps), you could check it out when you are available.

web3andrew commented:
@semuelle  Sir, any update?

semuelle commented on applications/Web3Box.md:
```diff
@@ -0,0 +1,137 @@
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can install and operate Web3Box client. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Article | We will publish an **article** in our official [Medium](https://medium.com/@web3box.official) that explains what was done/achieved as part of the grant.  |
+| 1. | **Dashboard**:Statistics and display of Polkadot users' assets | Statistics and display of the value of Polkadot users’ onchain assets, statistics and display of the value of input assets & output assets |
```
Where do these prices come from?

web3andrew replied:
> Where do these prices come from?

For on-chain assets data, we will get this info from the supported chains' mainnet in real time, like Polkadot Relaychain, Acala Parachain, etc. And for the token price data source, we will get the data from [chainlink](https://chain.link/) every 5 mins or 10mins. All data sources are requested in a decentralized way.

semuelle reviewed (commented):
Thank you for the update, @web3andrew, and sorry for the late reply. The application looks better now, but could you expand on what I raised in Q2? How is this sandbox implemented? How is it safer than a well-tested browser plugin?

> Sir，already updated our [Web3Box Github](https://github.com/web3box-labs/apps), you could check it out when you are available.

Could you explain what this project is? I tried building and running it, but neither worked. It looks like a template react app.

web3andrew commented:
> Thank you for the update, @web3andrew, and sorry for the late reply. The application looks better now, but could you expand on what I raised in Q2? How is this sandbox implemented? How is it safer than a well-tested browser plugin?
> 
> > Sir，already updated our [Web3Box Github](https://github.com/web3box-labs/apps), you could check it out when you are available.
> 
> Could you explain what this project is? I tried building and running it, but neither worked. It looks like a template react app.

@semuelle Firstly, we will provide the Mac & Win Web3Box Client for the Polkadot community, which could make community users' data saved locally and no interruptions from online attacks or viruses. 

And second,  to some degree, a well-tested browser plugin may be safer than the DApps in Web3Box, but the data is still controlled under centralized institutions, correct? Web3Box is a completely decentralized client with no control or surveillance of users' data.

And last, what Web3Box aims to be is trying to provide the Web3 world with a  much safer and more decentralized way to access data, DApps, and Web3 Projects.

BTW, you could run and build the app after we finished the Milestone1, we need more time to update and rich its details.

web3andrew commented:
> > And second, to some degree, a well-tested browser plugin may be safer than the DApps in Web3Box, but the data is still controlled under centralized institutions, correct? Web3Box is a completely decentralized client with no control or surveillance of users' data.
> 
> None of the wallets I know use a centralised service to store keys, and some allow you to use your own node. What centralised institutions do you mean?
> 
> > BTW, you could run and build the app after we finished the Milestone1, we need more time to update and rich its details.
> 
> The reason I asked about the project currently on Github was that we like to see if applicants have previous experience with building and maintaining open source software. A bunch of template code doesn't help with that. Do you have any other code or project material you can share?
@semuelle 
For centralized institutions, I mean that web2 products like Chrome Browser do control users' data and the users' data is under the surveillance of Chrome Browser. In contrast, Web3Box, Stand-alone & Decentralized Client , won't do that and will always allow users to keep data by themselves.

As for open source experience, our dev did participated in this defi project  https://github.com/netswap

web3andrew commented:
@semuelle Sir, any update about our application? :)

semuelle commented:
Thanks for the update, @web3andrew, and sorry for the late reply. There is now an even bigger gap between the project description and the deliverables. I think a desktop wallet should first and foremost focus on its core functionality, which is reflected in this milestone, but your project description contains a lot of other stuff that I'm not particularly keen on. Also, you haven't been able to show any previous open source work. Therefore, I am not going to support this application. However, I will share your application with my colleagues, who might feel different.

web3andrew commented:
> Thanks for the update, @web3andrew, and sorry for the late reply. There is now an even bigger gap between the project description and the deliverables. I think a desktop wallet should first and foremost focus on its core functionality, which is reflected in this milestone, but your project description contains a lot of other stuff that I'm not particularly keen on. Also, you haven't been able to show any previous open source work. Therefore, I am not going to support this application. However, I will share your application with my colleagues, who might feel different.

@semuelle Thanks, sir, we will wait for W3F finial decision about our application.

web3andrew commented:
> Thanks for the update, @web3andrew, and sorry for the late reply. There is now an even bigger gap between the project description and the deliverables. I think a desktop wallet should first and foremost focus on its core functionality, which is reflected in this milestone, but your project description contains a lot of other stuff that I'm not particularly keen on. Also, you haven't been able to show any previous open source work. Therefore, I am not going to support this application. However, I will share your application with my colleagues, who might feel different.

@semuelle  Thanks sir, according to your feedback, our team has simplified the proposal, and we will focus on Polkadot eco desktop dashboard, wallet & risk management. I have updated the proposal again. Hope you that your colleagues and you could read this latest [proposal](https://github.com/w3f/Grants-Program/pull/1256/commits/56b4f09099dfb423ad33fe564098f50ac4bfcb2f). BTW, we really care what you mentioned that "your project description contains a lot of other stuff that I'm not particularly keen on." , some other stuff is our version about Web3Box, and therefore we deleted them which are not related with current milestones.

Noc2 reviewed (changes requested):
Thanks for the application. Could you still sign the latest version of the CLA? (see CLAassistant). Apart from this, if I understand it correctly, you essentially want to develop a Electron-based wallet, correct? Could you integrate the technical details into the specification of the milestone deliverables? (e.g. react, electron, typescript? etc.) Also, you might want to take a look at the existing open-source wallets: https://wiki.polkadot.network/docs/build-open-source#user-interface and see if you can reuse some of them. Btw. https://github.com/polkadot-js/apps can also be run as a desktop electron wallet.

web3andrew commented:
BTW, We do love Polkadot. We are DOT holders. And we do need a chance to contribute more for Polkadot Community.

keeganquigley commented:
Thanks for the application @web3andrew and for the deep dive @semuelle, I'm afraid I have to agree with the assessment that I wouldn't support this application in its current state. The price is reasonable, but:

1. The deliverables still don't contain any specifics on the tech used, as @Noc2 mentioned. What programming language is being used? (Rust, TypeScript, JavaScript, etc.) What will the front end be built in? (React, Vue, etc.) What kind of encryption is being implemented? Are you going to utilize an API service such as [Subscan](https://support.subscan.io/#extrinsic)?

2. What does "risk assessment/management" mean exactly? The provided [screenshot](https://user-images.githubusercontent.com/117150831/200757322-7e71bb05-9246-4367-bf43-f006627e1bab.png) uses vague language and doesn't describe how the components work in detail.

3. The application has typos/formatting issues and vague language. For example, I don't believe "UGA platform" is an industry term that is widely used (I could be wrong). Another example is the [formatting](https://github.com/web3andrew/Grants-Program/blob/master/applications/Web3Box.md#team-members) of the team member section. Not that the language has to be perfect, but currently it gives me the impression that not much effort was put into it.

4. As mentioned there is still no proof of prior development experience.

5. Your description of a "privacy-safe sandbox concept" sounds like an air-gapped wallet such as Parity Signer. Is this the idea you are going for and if so are you planning to utilize QR codes, or implement some other methods of security?

6. Also, a note about the price feed oracle, to be competitive I think you would need to update it much more than every 5 or 10 minutes. Most desktop wallets are capable of updating at speeds that feel like real-time.

web3andrew commented:
> Thanks for the application @web3andrew and for the deep dive @semuelle, I'm afraid I have to agree with the assessment that I wouldn't support this application in its current state. The price is reasonable, but:
> 
>     1. The deliverables still don't contain any specifics on the tech used, as @Noc2 mentioned. What programming language is being used? (Rust, TypeScript, JavaScript, etc.) What will the front end be built in? (React, Vue, etc.) What kind of encryption is being implemented? Are you going to utilize an API service such as [Subscan](https://support.subscan.io/#extrinsic)?
> 
>     2. What does "risk assessment/management" mean exactly? The provided [screenshot](https://user-images.githubusercontent.com/117150831/200757322-7e71bb05-9246-4367-bf43-f006627e1bab.png) uses vague language and doesn't describe how the components work in detail.
> 
>     3. The application has typos/formatting issues and vague language. For example, I don't believe "UGA platform" is a widely used industry term (I could be wrong). Another example is the [formatting](https://github.com/web3andrew/Grants-Program/blob/master/applications/Web3Box.md#team-members) of the team member section. Not that the language has to be perfect, but currently it gives me the impression that not much effort was put into it.
> 
>     4. As mentioned there is still no proof of prior experience on open-source projects.
> 
>     5. Your description of a "privacy-safe sandbox concept" sounds like an air-gapped wallet such as [Parity Signer](https://github.com/paritytech/parity-signer). Is this the idea you are going for and if so are you planning to utilize QR codes, or implement some other methods with security in mind?
> 
>     6. Also, a note about the price feed oracle, to be competitive I think you would need to update it much more than every 5 or 10 minutes. Most desktop wallets are capable of updating at speeds that feel like real-time.

Q1:
Tech Details
Client language: Electron.js   React.js
Server language: Node.js
Build tool: Webpack

API Details
On-chain Data: https://polkadot.subscan.io/
Price Data(in real-time): https://chain.link/

Btw, Added them in the last commit https://github.com/w3f/Grants-Program/pull/1256/commits/6fa13deb30b1cd130b223da942ab0d370b51477f

Q2:
Added more descriptions about Risk Assessment in the latest commit https://github.com/w3f/Grants-Program/pull/1256/commits/6fa13deb30b1cd130b223da942ab0d370b51477f

Q3:
Deleted "UGA" and privacy-safe box descriptions, and we decided that we would focus on the dashboard, multi-chain wallet and risk assessment of Web3Box Client. Optimized the team info.

Q4:
Team members have participated in the defi project https://github.com/netswap, Netswap official website https://netswap.io/

Q5:
Focus on the dashboard, multi-chain wallet, and risk assessment of Web3Box Client. 

Q6:
Web3Box would get asset info from chainlink in real-time.

web3andrew commented:
@semuelle @Noc2 @keeganquigley  
Web3Box team made a big change about the proposal and we will focus on the implementation of milestone content such as Polkadot Dashboard, Multi-chain Wallet and Risk Assessment. We have streamlined the content of the Proposal by removing the UGA, privacy-safe box related descriptions. You guys could spare some time to read the latest commit again https://github.com/w3f/Grants-Program/pull/1256/commits/6fa13deb30b1cd130b223da942ab0d370b51477f 
Appreciate again for you guys’ feedback about Web3Box.

Noc2 reviewed (dismissed):
Sorry for the late reply here. I'm personally happy to go ahead with the application and will share it again with the rest of the team.

takahser commented on applications/Web3Box.md:
```diff
@@ -0,0 +1,112 @@
+Web3Box Dashboad Preview
+**![](https://lh3.googleusercontent.com/KZwZrmrRtQHMPqqcGkssfYtvC8SDmvGwB0WjbQ6r12Doqrx5yGyLcraGbVbqSFpYQjR-NcHUUK6tLYfmuq3huMCAsRHT0T_qiQ_AIV-KDvgroW1boEHsbeipCwXi245WDXsP3-LnI5BzpsVcXM4O6ux2RUrujJrY8Ah6qemsCDsxXz6FVLEtICp3JF79qw)**
+
+2. Risk assessment (follow-up grant)
+- Parachains Projects Scoring
+- Parechains Projects Risk Report
+- Parachains Abnormal warning
```
I believe @keeganquigley already asked about this, but I'm not quite clear how the risk assessment module would work. You're saying basically that it will rely on project scoring, risk report and abnormal warnings, but what do these terms actually mean and how would you source that data?

takahser reviewed (commented):
@web3andrew could you comment on how your wallet would differ from the [existing browser wallets](https://github.com/w3f/polkadot-wiki/edit/2630b90222b7d274ccabc57086faffa27c5a9f39/docs/build/build-open-source.md#user-interface)? I guess it's mainly the risk module, and for me it's not really clear how you'd evaluate these risks (see inline-comment).
Also, what do you mean by _Polkadot Ecology_?

web3andrew commented:
> _Polkadot Ecology_

@takahser @keeganquigley @Noc2 @semuelle 
For Polkadot Ecology, I mean this https://coinmarketcap.com/view/polkadot-ecosystem/ we would provide the risk assessment for Polkadot Ecology projects in the above coinmarketcap list. And the basic functions, the Web3Box Polkadot Dashboard & Wallet would be related to some risk assessment linkage of Polkadot assets, contracts, and accounts, which was planned to do about personal risk assessment in the future. So wallet and risk assessment are not independent.

And now, for risk assessment, we are focusing on project risk assessment in Polkadot Ecology. The initial evaluation dimensions are as follows and will be described in more detail by follow-up grant, and finally, each project will get its own risk score according to the following evaluation dimensions.

Risk Type 1: Projects' Rug-pull
-Major holders' wallet assets distribution  (Data Source: blockchain explorer or other centralized third-party)
-Market Cap/ TVL (Data Source: Coinmarketcap)
-Twitter followers increasing rate  (Data Source: official Twitter )
-GitHub updating rate  (Data Source: official Github)

Risk Type 2: Projects' Attacking
-Audit Report (Data Source: audit institutions, like Certik, Slowmist, etc)
-Security Vulnerability (Data Source: audit institutions, like Certik, Slowmist, etc)

Risk Type 3: Related to other highly risky projects
-Investment and financing report  (Data Source: pitchbook)
-Cooperations with other projects (Data Source: official Twitter)

Risk Type 4: Users' Crash
-Token price volatility ( Data Source: Chainlink/Coinmarketcap)
-Rate of change in transactions volume ( Data Source: blockchain explorer)
-User sentiment index  ( Data Source: our devs will use an AI algorithm to figure out it basing on the comments of projects' Twitter)

web3andrew commented:
> > _Polkadot Ecology_
> 
> @takahser @keeganquigley @Noc2 @semuelle For Polkadot Ecology, I mean this https://coinmarketcap.com/view/polkadot-ecosystem/ we would provide the risk assessment for Polkadot Ecology projects in the above coinmarketcap list. And the basic functions, the Web3Box Polkadot Dashboard & Wallet would be related to some risk assessment linkage of Polkadot assets, contracts, and accounts, which was planned to do about personal risk assessment in the future. So wallet and risk assessment are not independent.
> 
> And now, for risk assessment, we are focusing on project risk assessment in Polkadot Ecology. The initial evaluation dimensions are as follows and will be described in more detail by follow-up grant, and finally, each project will get its own risk score according to the following evaluation dimensions.
> 
> Risk Type 1: Projects' Rug-pull -Major holders' wallet assets distribution (Data Source: blockchain explorer or other centralized third-party) -Market Cap/ TVL (Data Source: Coinmarketcap) -Twitter followers increasing rate (Data Source: official Twitter ) -GitHub updating rate (Data Source: official Github)
> 
> Risk Type 2: Projects' Attacking -Audit Report (Data Source: audit institutions, like Certik, Slowmist, etc) -Security Vulnerability (Data Source: audit institutions, like Certik, Slowmist, etc)
> 
> Risk Type 3: Related to other highly risky projects -Investment and financing report (Data Source: pitchbook) -Cooperations with other projects (Data Source: official Twitter)
> 
> Risk Type 4: Users' Crash -Token price volatility ( Data Source: Chainlink/Coinmarketcap) -Rate of change in transactions volume ( Data Source: blockchain explorer) -User sentiment index ( Data Source: our devs will use an AI algorithm to figure out it basing on the comments of projects' Twitter)

@takahser @keeganquigley @Noc2 @semuelle   Hi, guys, any update? :)

Noc2 commented:
Thanks for pinging us here. I will share the application again with the rest of the team.

keeganquigley commented:
Hi @web3andrew thanks for the updates and the additional detailed info. However, I think you can remove the 2. [Risk Assement](https://github.com/w3f/Grants-Program/pull/1256/files#diff-8729d41b49c15a6dda89f683392c6fa489dfbf08a1327dd456107f778c3935cbR26-R29) part since this module will be part of a follow-up grant, so we'll worry about it then.

Regarding your deliverables, 0e. [Tech Details](https://github.com/w3f/Grants-Program/pull/1256/files#diff-8729d41b49c15a6dda89f683392c6fa489dfbf08a1327dd456107f778c3935cbR90) is not something we can evaluate, so can you remove this from the deliverables? Deliverables basically need to be tangible units of work that we can test and evaluate. 

For 0f. [API Details](https://github.com/w3f/Grants-Program/pull/1256/files#diff-8729d41b49c15a6dda89f683392c6fa489dfbf08a1327dd456107f778c3935cbR91) can you change this to describe how you will implement the API and what the final implementation might look like?

Considering the reasonable price for this grant, if you can fix the deliverables I will be in a better position to approve. Thanks!

web3andrew commented:
@keeganquigley Hi, Keegan, I have updated the [MD File](https://github.com/w3f/Grants-Program/pull/1256/commits/b6bd8e139792bd83599ff1384c54d3387cd69434) according to your feedback, you could check it out. @Noc2 Hi, David, when I submitted the latest commit, Github dismissed your stale review automatically, but I did't know why it happened. Could you mark "APPROVED" again when you are available, sorry about the inconvenience.

Noc2 reviewed (approved):
Sure. Every time there is an update, I need to re-approve it and double-check that you haven't secretly increased the price, for example ;-)

web3andrew commented:
> Sure. Every time there is an update, I need to re-approve it and double-check that you haven't secretly increased the price, for example ;-)

Got it, I am not using Github very often since I am not a coder(But I am the Web3 builder, all in Web3 now). To be honest, I was very scared at that moment.

keeganquigley commented:
Thanks for the additional changes @web3andrew, I'm willing to give you a chance and approve the application now! Congratulations and I look forward to seeing what you can come up with.

web3andrew commented:
> Thanks for the additional changes @web3andrew, I'm willing to give you a chance and approve the application now! Congratulations and I look forward to seeing what you can come up with.

@Noc2 @keeganquigley Thanks, David & Keegan, appreciate you guys that giving us this chance and trusting Web3Box team, we will try our best to complete milestones and keep building Web3.

web3andrew commented:
<img width="1545" alt="image" src="https://user-images.githubusercontent.com/117150831/217862468-e3fa44d7-1e7d-4793-ad01-f77e3fc83d97.png">
@semuelle @noc @keeganquigley  Hi, guys, what should I do with this? Unfortunately, I submitted our milestone delivery unsuccessfully.I have writen the address, contact name, contact email, and total cost in this latest application md b6bd8e139792bd83599ff1384c54d3387cd69434.
<img width="732" alt="image" src="https://user-images.githubusercontent.com/117150831/217864412-249422eb-5e2e-425e-9c6e-587bae6f2f0a.png">
<img width="405" alt="image" src="https://user-images.githubusercontent.com/117150831/217864469-f3cf0b11-8607-4396-95b8-520f1469c534.png">
<img width="492" alt="image" src="https://user-images.githubusercontent.com/117150831/217864731-d19f8795-6ea4-4102-a619-5ab7a0b5a0b7.png">
